### PR TITLE
Add JAVA_TOOL_OPTIONS env to task template

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Add JAVA_TOOL_OPTIONS env to task template to account accomodate for the task's resource limit.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [X] Any pending items have an associated Issue

## Complete description
Propagate the Resource Memory limit into env `JAVA_TOOL_OPTIONS=-Xmx` in the task template so that the jvm running the java task picks this up and the the upper limit accordingly.

As described in the [java documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html), `JAVA_TOOL_OPTIONS` is picked up by the `java` command to inject options as if they were written in the command line.

The parameter `-Xmx<size>[g|G|m|M|k|K]` described [here](https://docs.oracle.com/cd/E13150_01/jrockit_jvm/jrockit/jrdocs/refman/optionX.html) is used to specify the max size for the memory heap.

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/2285

## Follow-up issue
https://github.com/flyteorg/flyte/issues/2286

### CODE REVIEW

File: jflyte/src/main/java/org/flyte/jflyte/ProjectClosure.java [hunk 1]
```diff
@@ -426,6 +410,15 @@ static TaskTemplate createTaskTemplateForRunnableTask(RunnableTask task, String
         .build();
   }
 
+  private static Optional<KeyValuePair> javaToolOptionsEnv(Resources resources) {
+    Map<ResourceName, String> limits = resources.limits();
+    if (limits == null || !limits.containsKey(ResourceName.MEMORY)) {
+      return Optional.empty();
+    }
+    String javaQuantity = asJavaQuantity(limits.get(ResourceName.MEMORY));
```
Comment (ckiosidis) [inline jflyte/src/main/java/org/flyte/jflyte/ProjectClosure.java:418]:
```suggestion
    String maxMemory = asJavaQuantity(limits.get(ResourceName.MEMORY));
```

File: jflyte/src/main/java/org/flyte/jflyte/ProjectClosure.java [hunk 2]
```diff
@@ -426,6 +410,15 @@ static TaskTemplate createTaskTemplateForRunnableTask(RunnableTask task, String
...
+    return Optional.of(KeyValuePair.of("JAVA_TOOL_OPTIONS", "-Xmx" + javaQuantity));
```
Comment (ckiosidis) [inline jflyte/src/main/java/org/flyte/jflyte/ProjectClosure.java:419]:
```suggestion
    return Optional.of(KeyValuePair.of("JAVA_TOOL_OPTIONS", "-Xmx" + maxMemory));
```

**Review by ckiosidis — APPROVED** [review]
looks good and clean. 
It's good to mention the steps taken to test this change for future reference.
Also we should not forget to create an issue to expand both the env and the JAVA_TOOL_OPTIONS and make them configurable to the user

**Comment by narape** [comment]
I created https://github.com/flyteorg/flyte/issues/2286 and I added to the description
